ERCS/erc-7683.md

```diff
@@ -24,24 +24,34 @@ By implementing a standard, cross-chain intents systems can interoperate and sha
 
 The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
 
-### CrossChainOrder struct
+### Glossary of Terms
 
-A compliant cross-chain order type MUST be ABI decodable into the `CrossChainOrder` type.
+- **Destination Chain**: the chain where the intent is executed and the user recieves funds. Note: intents can involve multiple destination chains.
+- **Filler**: a participant who fulfils a user intent on the destination chain(s) and recieves payment as a reward.
+- **Leg**: a portion of the user intent that can be executed independently from others. All legs must be executed for an intent to be considered fulfilled.
+- **Origin chain**: the chain where the user sends funds.
```

Comment:
would this system allow for multiple origin chains?

Reply:
No, it wouldn't. The current standard assumes that there's only a single origin chain per order. These can be composed just by combining multiple 7683 orders on multiple origins into a larger construct.

Thoughts? Would this cause problems for your use case?

Reply:
The main problem I see is that it is unclear in this case which one of these independent orders should actually execute the calls on the target chain - especially in cases where it isn't clear which order would be able to be settled first. There might be ways of solving this problem outside of the protocol, but will have a think on this

Reply:
@mrice32 Not allowing multiple source chains in the same Order struct will become problematic for our use case, and most probably many other smart account systems too.
A nice feature of smart accounts is they can allow selective replay of the same signature across different chains if the user wants.
So a user who is pulling funds from many different source chains and depositing to a single target chain would only need to sign one order, which would contain the chainIds of all of the source chains, and the smart account would just parse the order and check if the current chain ID matches any of the IDs signed in the order.

But to do this, the hash of the order should be the same for all chains. This can only be achieved if the order follows a pattern like - 
```
ChainOrder { 
uint256 originChainId
uint256 targetChainId
address originSettler
address targetContract
... }

Order {
ChainOrder[] orders
}
```
In this simplified example, the hash of the order could be the same for all origin chains, so the end-user only h as to make 1 signature for N chains.
There are some tricks which would let EOAs utilize this UX improvement too.

It seems like 7683 could support this pretty easily, by just packing all "per chain" information into a struct and then creating an array of these structs for the gaslessOrder and resolvedOrder.

I don't think this would add much gas or complexity to the current implementation.

Reply:
Okay, this is very interesting. A few questions:
1. Why can't you just have a list of transactions that each contain an ERC-7683 open function calldata (order) that you want to send? And wouldn't this be more general anyway? Something like:
```sol
struct Transaction {
  address to;
  value: uint256;
  callData: bytes;
  chainId: uint256;
}

struct MultichainCall {
  Transaction[] transactions;
  // Per chain replay protection.
  nonce: uint256;
}

// Call this on your smart account:
function multichainCall(MultichainCall memory multichainCall, bytes memory signature) {
  // Check if nonce is used on this chain.
  // Walk through all txns and execute any that have chainId == block.chainId.
}
```

2. Is atomicity the issue? Note: the only functional difference I see in 7683 allowing multi-origin within the standard vs having users of the standard compose multiple orders to create a multi-origin order is per-destination atomicity. Out of the box, I can't guarantee that either all the orders go through or none of them do. But you may be able to do this through creating conditional relationships between the fills of different orders, but that can get complicated.